ffi/src/input.rs

```diff
@@ -1,11 +1,131 @@
 #[diplomat::bridge]
 pub mod ffi {
+    use crate::pdu::ffi::FastPathInputEventIterator;
+
     #[diplomat::opaque]
     pub struct InputDatabase(pub ironrdp::input::Database);
 
     impl InputDatabase {
         pub fn new() -> Box<InputDatabase> {
             Box::new(InputDatabase(ironrdp::input::Database::new()))
         }
+
+        pub fn apply(&mut self, operation: &Operation) -> Box<FastPathInputEventIterator> {
+            let res = self.0.apply(std::iter::once(operation.0.clone()));
+            Box::new(res.to_vec().into())
+        }
+    }
+
+    #[diplomat::opaque]
+    pub struct Operation(pub ironrdp::input::Operation);
+
+    pub enum OperationType {
+        MouseButtonPressed,
+        MouseButtonReleased,
+        MouseMove,
+        WheelRotations,
+        KeyPressed,
+        KeyReleased,
+        UnicodeKeyPressed,
+        UnicodeKeyReleased,
+    }
+
+    #[diplomat::opaque]
+    pub struct MousePosition(pub ironrdp::input::MousePosition);
+
+    impl MousePosition {
+        pub fn new(x: u16, y: u16) -> Box<MousePosition> {
+            Box::new(MousePosition(ironrdp::input::MousePosition { x, y }))
+        }
+
+        pub fn as_operation(&self) -> Box<Operation> {
```

Comment:
naming: `as_move_operation`
```suggestion
        pub fn as_move_operation(&self) -> Box<Operation> {
```